Makefile

```
@@ -9,8 +9,8 @@ clear-cache:
 
 # generates mocks of interfaces for testing
 .PHONY: genmocks
-genmocks:
-	@command -v mockgen >/dev/null || go install github.com/golang/mock/mockgen
+genmocks: go.mod go.sum
+	go install go.uber.org/mock/mockgen
 	go generate ./...
```

```suggestion
genmocks:
	go generate -run mockgen ./...
```

I think this is a better way to express this:

1. dependencies on a phony target don't make a lot of sense
1. all the `generate` directives should be using `go run`
1. the `go generate` command should only actually run `mockgen`